core/trino-main/src/main/java/io/trino/operator/join/SortedPositionLinks.java

```diff
@@ -45,9 +48,12 @@
     public static class FactoryBuilder
             implements PositionLinks.FactoryBuilder
     {
-        private final Int2ObjectMap<IntArrayList> positionLinks;
+        // Cache lambda instance for use with createIfAbsent
+        private static final IntFunction<IntArrayList> NEW_INT_LIST = (ignored) -> new IntArrayList();
```

Comment:
I thought stateless lambdas are cached by the jvm. Does this make a performance difference?

Reply:
You are correct, and I'm realizing now that this change is a hold-over from when I was looking at a different version of this that had an overloaded alternative to `computeIfAbsent(int, IntFunction<?>)` called `computeIfAbsent(int, Int2ObjectFunction<?>)` which I was trying to understand the relationship of. I agree, this caching is awkward and unnecessary so I've removed it.